skills/diagnose_system_lag/scripts/diagnose.sh

```
@@ -0,0 +1,339 @@
+    SWAP_TOTAL=$(echo "$FREE_OUTPUT" | grep -i '交换\|swap:' | awk '{print $2}')
+    SWAP_USED=$(echo "$FREE_OUTPUT" | grep -i '交换\|swap:' | awk '{print $3}')
+    SWAP_FREE=$(echo "$FREE_OUTPUT" | grep -i '交换\|swap:' | awk '{print $4}')
+
+    # 判定内存状态
+    MEMORY_CRITICAL=false
+    SWAP_CRITICAL=false
+
+    # 检查 Swap 使用情况
+    if [[ "$SWAP_TOTAL" != "0" && "$SWAP_TOTAL" != "0B" ]]; then
+        # 提取数字和单位
+        SWAP_USED_NUM=$(echo "$SWAP_USED" | sed 's/[^0-9.]//g')
```

Shellcheck (0.11.0)

[style] See if you can use ${variable//search/replace} instead.

(SC2001)